builtin/rebase.c

```diff
@@ -470,11 +470,15 @@ static int reset_head(struct object_id *oid, const char *action,
 		strbuf_addstr(&msg, "updating HEAD");
 		reflog_head = msg.buf;
 	}
-	if (!switch_to_branch)
+	if (!switch_to_branch) {
 		ret = update_ref(reflog_head, "HEAD", oid, orig,
 				 detach_head ? REF_NO_DEREF : 0,
 				 UPDATE_REFS_MSG_ON_ERR);
-	else {
+		if (!strcmp("checkout", action))
```

Comment:
Better use a new flag, `RESET_HEAD_RUN_POST_CHECKOUT_HOOK` (or maybe `RESET_HEAD_RUN_HOOK`, as it is pretty obvious which one to run).?

Reply:
Oh, and I don't think this should be guarded by `if (!switch_to_branch)` but simply run after the `else` clause.

Reply:
There are 2 calls with "checkout" action - one when starting the rebase, and another one when it is done, and checks out the branch. On the second invocation the commit is the same (old and new), and the previous behavior (I think) was to only run the hook on the first checkout.

Reply:
> Better use a new flag

Done